src/RequestAuthenticator.php

```diff
@@ -71,7 +71,7 @@ public function authenticate(RequestInterface $request)
         $compareSignature = $compareAuthHeader->getSignature();
 
 
-        if ($signature !== $compareSignature) {
+        if (!hash_equals($signature, $compareSignature)) {
```

Comment:
Looks like it's not a problem currently; the implementor added that to ensure forward compatibility in case the function implementation changed in the future: http://stackoverflow.com/a/27912885 But I've gone ahead and fixed it. Thanks!